### ISSUE TITLE
Add list method for APIObjects

### ISSUE THREAD

**Issue by benedikt-bartscher** [kr8s-org/kr8s#249]
### Which project are you requesting an enhancement for?

kr8s

### What do you need?

Hey @jacobtomlinson, thank you very much for this awesome project, I really enjoy using it!

For getting custom APIObject I prefer this syntax:
```
test: MyApiObject = await MyApiObject.get(name="test")
```
over this syntax
```
test: List[MyApiObject] = await kr8s.asyncio.get("myapiobject", name="test")
```
because it is typed stronger and the results are instances of the correct class.
However, there is no way to list/get multiple objects with the first syntax, or am I missing something?

**Comment by jacobtomlinson** [comment]
I think it would be minimal effort to support something like this.

```python
test: List[MyApiObject] = await MyApiObject.list()
```

Another thing we could do is allow `kr8s.asyncio.get` to take in the class and avoid the lookup, although the return type will still be the `APIObject` base class.

```python
test: List[MyApiObject] = await kr8s.asyncio.get(MyApiObject)
```

**Comment by jacobtomlinson** [comment]
@benedikt-bartscher I went ahead and released `v0.12.8` which has the `MyApiObject.list()` method implemented.

**Comment by benedikt-bartscher** [comment]
Thanks, that was a quick one. I just tested the new `list()` api and it works fine.
